nixos/modules/services/development/hoogle.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+# services.hoogle = {
+#   enable = true;
+#   packages = hp: with hp; [ text lens ];
+#   haskellPackages = pkgs.haskellPackages;
+# };
+
+with lib;
+
+let
+
+  cfg = config.services.hoogle;
+  ghcWithHoogle = pkgs.haskellPackages.ghcWithHoogle;
+
+in {
+
+  options.services.hoogle = {
+    enable = mkEnableOption "Hoogle Documentation service";
+
+    port = mkOption {
+      type = types.int;
+      default = 8080;
+      description = ''
+        Port number Hoogle will be listening to.
+      '';
+    };
+
+    packages = mkOption {
+      default = hp: [];
+      example = "hp: with hp; [ text lens ]";
+      description = ''
+        A function that takes a haskell package set and returns a list of
```

Comment:
I think it's better to focus on what the option is supposed to accomplish, i.e., specifying the packages available to the documentation server, and then say that it's a function that takes the haskellPackages attset &c